### ISSUE TITLE
Multi-statement simple query should run in an implict transaction block

### ISSUE THREAD

**Issue by FiV0** [xtdb/crux#3803]
The following currently returns both ids.
```clj
(deftest multistatement-simple-query-should-run-in-implicit-transaction-block
  (with-open [conn (jdbc-conn)]
    (jdbc/execute! conn ["BEGIN;
                          INSERT INTO docs(_id) VALUES (1);
                          COMMIT;
                          INSERT INTO docs(_id) VALUES (2);
                          SELECT 0/1"])

    (t/is (= [{:id 1}] (jdbc/execute! conn ["SELECT * FROM docs"])))))
```
see https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-MULTI-STATEMENT

**Comment by refset** [comment]
~~Dupe of https://github.com/xtdb/xtdb/issues/3242 (I think? if so, I don't mind which gets closed)~~

**Comment by FiV0** [comment]
The other issue sounds more like we don't support statements separated by `;` which the test clearly shows we do. I think this is more about the semantics of multi-statement simple queries which is currently not correctly implemented.

**Comment by jarohen** [comment]
@FiV0 I know this is what the test says, but I can't see how we could currently support `;`-delimited queries - I wonder if the JDBC client is splitting them up?

edit: yep, we're receiving them as separate statements

**Comment by jarohen** [comment]
Thanks for raising this, @FiV0 - I've included this as the spec for #3242.
